src/fetchers/tarball-fetcher.js

```diff
@@ -329,3 +350,16 @@ export class LocalTarballFetcher extends TarballFetcher {
     return this.fetchFromLocal(this.reference);
   }
 }
+
+type UrlParts = {
```

Comment:
This is copied from https://github.com/yarnpkg/yarn/blob/b8a565e562810f9bf86f3ac0531ad348ed7e9d0a/src/registries/npm-registry.js#L77-L88

I would like to put it in a more central location. I ask the Yarn team where would the most appropriate destination? 😄